### DISCUSSION TITLE
Why does the generated HTML end with a new line?

### DISCUSSION THREAD

**Discussion by photino** [kivikakk/comrak#451] [Q&A] [answered]
In `comrak`, the input `## Math` will generate `<h2>Math</h2>\n`. Why does the generated HTML end with a new line? It is inconsistent with other lib like `markdown`.

**Comment by kivikakk** [comment, marked as answer]
This is how upstream [`cmark`](https://github.com/commonmark/cmark), [`cmark-gfm`](https://github.com/github/cmark-gfm), etc. work, including the automated spec tests which expect a newline at the end of the output. In other words, it's consistent with the CommonMark ecosystem, which Comrak is very much a part of.
